src/priorityq.py

```diff
@@ -0,0 +1,44 @@
+# -*- coding: utf-8 -*-
+"""File implements a priority queue data structure in python."""
+from __future__ import unicode_literals
+from heap import Heap
+
+
+class Priorityq(object):
+    """Class implements a priority queue data structure in Python."""
+    order = 0
+
+    def __init__(self, iterable):
+        """Init instance of priority queue class,
+        iterate through data if provided as an argument."""
+        pq_list = []
+        if type(iterable) is not list:
+            raise IndexError("Enter a list of tuples, each with 2 values.")
+        for i in iterable:
+            if type(i) is not tuple:
+                raise IndexError("Enter a list of tuples, each with 2 values.")
+            elif type(i[0]) is not int:
+                raise IndexError("First value in tuple must be an integer.")
+            self.order += 1
+            new_tuple = (i[0], self.order, i[1])
+            (pq_list).append(new_tuple)
+        self.heap = Heap(pq_list)
+
+    def insert(self, tup):
```

Comment:
It would be more user friendly to allow user to pass 2 args, value and priority, rather than requiring a tuple data type only.